js/modules/gomodule.go

```diff
@@ -25,10 +25,22 @@ func (gm *goModule) Instantiate(rt *sobek.Runtime) (sobek.CyclicModuleInstance,
 	if gm.exportedNames == nil {
 		named := mi.Exports().Named
 
-		gm.exportedNames = make([]string, len(named))
-		for name := range named {
-			gm.exportedNames = append(gm.exportedNames, name)
+		switch {
+		case len(named) == 0 && mi.Exports().Default != nil:
+			// If named length is 0 but default is defined, then try to work with
+			// default and extract the names of the object's properties. This
+			// behavior isn't ESM compatible, but we do want to allow defaults to
+			// be imported as namespaced object, which is also how node works.
+			if obj, ok := mi.Exports().Default.(*sobek.Object); ok {
+				gm.exportedNames = obj.GetOwnPropertyNames()
+			}
+		default:
+			gm.exportedNames = make([]string, len(named))
```

Comment:
I guess this is from the original code, but it is ... wrong :sweat_smile: As it first make a slice with the required size and then add more elements to the end

https://go.dev/play/p/W66MinRKLc3

This is doubtful to be an issue, as ... well empty strings aren't going to be useful but still
```suggestion
			gm.exportedNames = make([]string, 0, len(named))
```

Reply:
Fixed in 134399b4273ecd631c6fa41bc0a91cec1623c827